griffin/secret_management_util.py

```diff
@@ -78,15 +79,33 @@ def get_secret_from_vault(self, secret_key):
 
     def get_secret_from_cache(self, secret_key):
         cache_key = construct_cache_key(secret_key)
-        cache_value = self.redis_connection.get(cache_key)
+        cache_value = self.cache.get(cache_key)
         if cache_value is None:
             raise SecretNotFoundException(f"Secret Unavailable in cache for secret key - {secret_key}")
-        return cache_value.decode("utf-8")
+        LOGGER.info(f'Secret Value fetched from cache for secret key {secret_key}')
+        return cache_value
 
     def set_secret_to_cache(self, key, value):
```

Comment:
@suprithak16 lets also please add a utility function to delete an item from cache. Dont have a concrete use case as of now, but we might need it